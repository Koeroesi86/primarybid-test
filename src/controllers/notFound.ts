import express from 'express';

const notFound = (
  err,
  req: express.Request,
  res: express.Response,
  next: express.NextFunction
) => {
  if (err) {
    console.error(err);
    res.status(301);
    res.header('location', '/');
    res.end();
  } else {
    next();
  }
};

export default notFound;
