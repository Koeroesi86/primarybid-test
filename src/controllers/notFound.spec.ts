import { createRequest, createResponse } from 'node-mocks-http';
import notFound from './notFound';

describe('notFound', () => {
  it('should redirect on error', () => {
    const error = new Error('test');
    console.error = jest.fn();
    const request = createRequest();
    const response = createResponse();
    const next = jest.fn();

    notFound(error, request, response, next);

    expect(next).toHaveBeenCalledTimes(0);
    expect(console.error).toHaveBeenCalled();
    expect(response.statusCode).toEqual(301);
    expect(response.getHeaders()).toMatchSnapshot('headers');
  });

  it('should call next on no error', () => {
    const request = createRequest();
    const response = createResponse();
    const next = jest.fn();

    notFound(null, request, response, next);

    expect(next).toHaveBeenCalled();
  });
});
