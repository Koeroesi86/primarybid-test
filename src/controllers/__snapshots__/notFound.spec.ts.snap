// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notFound should redirect on error: headers 1`] = `
Object {
  "location": "/",
}
`;
